messaging/src/main/java/nl/nn/adapterframework/extensions/kafka/KafkaSender.java

```diff
@@ -0,0 +1,108 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.extensions.kafka;
+
+import java.util.concurrent.Future;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.ISender;
+import nl.nn.adapterframework.core.PipeLineSession;
+import nl.nn.adapterframework.core.SenderException;
+import nl.nn.adapterframework.core.SenderResult;
+import nl.nn.adapterframework.stream.Message;
+
+/**
+ * Experimental {@link ISender} for sending messages to a Kafka instance.
+ */
+@Log4j2
+public class KafkaSender extends KafkaFacade implements ISender {
+
+	//setter is for testing purposes only.
+	private @Setter(AccessLevel.PACKAGE) Producer<String, byte[]> producer;
+
+	/** The topic to send messages to. Only one topic per sender. Wildcards are not supported. */
+	private @Setter String topic;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+		if (StringUtils.isEmpty(topic))
+			throw new ConfigurationException("topic must be specified");
+		if (topic.contains(","))
+			throw new ConfigurationException("Only one topic is allowed to be used for sender.");
+		if (topic.contains("*"))
+			throw new ConfigurationException("Wildcards are not allowed to be used for sender.");
+	}
+
+	@Override
+	public void open() throws SenderException {
+		producer = new KafkaProducer<>(properties, new StringSerializer(), new ByteArraySerializer());
+		try {
+			Thread.sleep(100);
+		} catch (InterruptedException e) {
+			Thread.currentThread().interrupt();
+			throw new SenderException(e);
+		}
```

Comment:
Is dit echt nodig? Ik vind dit een beetje lelijk...

Reply:
Tsja, zo is de "interrupted" flag van het thread wel weer gezet. Dat zou, voor zover ik de documentatie begrijp, niet opnieuw moeten throwen.

Zo is Sonar ook weer blij.